### PR TITLE
Auto clear

### PR DESCRIPTION
**Pull request by jpbarends**
This should solve the auto-clear from your to-do list.
Also redraws the board after a change of the standard level.

### CODE REVIEW

File: js/MineSweeper.js [hunk 1]
```diff
@@ -50,28 +50,31 @@ jQuery(function ($) {
         STATE_FLAGGED = 'flagged',
         STATE_EXPLODE = 'explode',
         STATE_QUESTION = 'question';
+	var LEFT_MB=1,
```
Comment (michaelbutler) [inline js/MineSweeper.js]: can you change this to LEFT_MOUSE_BUTTON and RIGHT_MOUSE_BUTTON?

File: js/MineSweeper.js [hunk 2]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
         };
 
-        this.initWorkers = function () {
+		this.callWorker = function(task,par) {
+            $('.ajax-loading').removeClass('invisible');
+            var job = {
+                type: task, // message type
+                grid: msObj.grid
+            };
+			if (typeof par === 'number') {
```
Comment (michaelbutler) [inline js/MineSweeper.js]: Can you fix the indenting on these lines? It looks misaligned. I think the standard should be 4 spaces per indent

File: js/MineSweeper.js [hunk 3]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
+				job.mines=par;
+			}
+			else if (typeof par === 'object'){
+				job.x=par.x;
```
Comment (michaelbutler) [inline js/MineSweeper.js]: please put spaces around all '=' and math operators llike '+' and '-'

File: js/MineSweeper.js [hunk 4]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
+				job.y=par.y;
+			}
+            msObj.worker.postMessage(JSON.stringify(job));
```
Comment (michaelbutler) [inline js/MineSweeper.js]: fix indent

File: js/MineSweeper.js [hunk 5]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
-            self.element.on('mouseup', function (ev) {
-                if (ev.which === 3) {
-                    self.RIGHT_BUTTON_TIMEOUT = setTimeout(function () {
-                        self.RIGHT_MOUSE_CLICKED = false;
+            msUI.on('mouseup', function (ev) {
+                if (ev.which === RIGHT_MB) {
+                    msObj.RIGHT_BUTTON_TIMEOUT = setTimeout(function () {
+                        msObj.RIGHT_MOUSE_DOWN = false;
+                    }, 50);
+				}
```
Comment (michaelbutler) [inline js/MineSweeper.js]: indent messed up here

File: js/MineSweeper.js [hunk 6]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
+						var x = targ.attr('data-x')-1;
+						var ud=targ.parent().prev();
+						for(var i=x;i<x+3;i++)
+							ud.children(".unknown.[data-x="+i+"]").addClass('test');
+						targ.prev('.unknown').addClass('test');
+						targ.next('.unknown').addClass('test');
+						ud=targ.parent().next();
+						for(var i=x;i<x+3;i++)
+							ud.children(".unknown.[data-x="+i+"]").addClass('test');
+				}
```
Comment (michaelbutler) [inline js/MineSweeper.js]: could you fix all indents and spacing for the above 10+ lines

File: js/MineSweeper.js [hunk 7]
```diff
@@ -79,86 +82,127 @@ jQuery(function ($) {
...
+            msUI.on('mousedown','.cell', function (ev) {
                 var targ = $(ev.target);
-                ev.preventDefault();
-                self.handleRightClick(targ);
+                if ((ev.which === LEFT_MB&&msObj.RIGHT_MOUSE_DOWN)||
+					(ev.which === RIGHT_MB&&msObj.LEFT_MOUSE_DOWN)) {
+						var x = targ.attr('data-x')-1;
+						var ud=targ.parent().prev();
+						for(var i=x;i<x+3;i++)
+							ud.children(".unknown.[data-x="+i+"]").addClass('test');
```
Comment (michaelbutler) [inline js/MineSweeper.js]: Please use { and } even if it is only 1 line (for readability)

File: js/cell_toucher.js [hunk 8]
```diff
@@ -19,70 +19,31 @@
...
 
 /**
  * @param obj
  * @param grid
  * @returns {Array}
  */
-function get8AdjacentSquares(obj, grid) {
-    var array = [],
+function getAdjacentSquares(obj, grid) {
+	    var Gij,
```
Comment (michaelbutler) [inline js/cell_toucher.js]: could you change Gij name to somthing more readable. Good job reducing the code in this function though.

File: js/cell_toucher.js [hunk 9]
```diff
@@ -98,77 +60,50 @@ function get8AdjacentSquares(obj, grid) {
...
-                closed_cells += 1;
+function minesweeperCalculateWin(grid,mines) {
+    var closed_cells = 0,
+        cell;
+
+    for (var y = 0; y < grid.length; y++) {
+        for (var x = 0; x < grid[0].length; x++) {
+            cell = grid[y][x];
+            if (!(cell.state === 'open' || cell.state === 'number')) {
+                closed_cells++;
```
Comment (michaelbutler) [inline js/cell_toucher.js]: please use += instead of ++

**Comment by michaelbutler** [comment]
Please fix up the code formatting so it matches with the rest of the code. Mainly, 4 spaces for indents instead of tab characters (\t), spaces around math or assignment operators, spaces to the right of commas, and requiring { and } around every flow control like for, while, if/else blocks.

**Comment by jpbarends** [comment]
Should look better now.

**Comment by michaelbutler** [comment]
@jpbarends Thanks again. Finally got to test out the left + right mouse button autoclear feature. Really like it.

**Comment by jpbarends** [comment]
You're welcome. 
I since completely reworked the program, mostly to get to know javascript and jQuery better. All status info is now stored in the document object (if that's the right terminology) so the syncing of it and the grid array is no longer needed. I could upload it if you're curious.

**Pull request merged**